presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -191,6 +193,11 @@ public Type getCoercion(Expression expression)
         return coercions.get(expression);
     }
 
+    public boolean isTypeOnlyCoercion(Expression expression)
```

Comment:
It is really hard to separate those 2 commits now. And actually keeping those changes in separate is confusing. So it is hard to understand why those changes are being made. Instead of separating this commit into 2 I'm just going to change the commit message.